cli/src/base_cli/commands/listen.rs

```diff
@@ -53,110 +51,40 @@ impl ListenCommand {
 				}
 			};
 
-			let event_results = subscription.next_events::<RuntimeEvent, Hash>().unwrap();
+			let maybe_event_results_solo =
+				subscription.next_events::<solochain::RuntimeEvent, solochain::Hash>();
+			let maybe_event_results_para =
+				subscription.next_events::<parachain::RuntimeEvent, parachain::Hash>();
+
 			blocks += 1;
-			match event_results {
-				Ok(evts) =>
+			match maybe_event_results_solo {
+				Some(Ok(evts)) => {
```

Comment:
this is now the pattern how we support events from both solo and parachains:
1. try to decode as solo
2. if it fails, try to decode as para

this leads to quite some redundancy and I'd like to see macros here at some point. But the problem of solving all use cases with macros isn't trivial and we may just want to serialize events to their Debug fmt and then do String ops on them to match modules, events, fields

Reply:
darn. like this, the parachain only sees every second block because the first is swallowed by the failed solo::next_events

Reply:
looks like we should use this anyway
https://github.com/scs/substrate-api-client/blob/02bfcefc7e34f0583b8e617c6e966aa10a212ee5/testing/examples/events_tests.rs#L91